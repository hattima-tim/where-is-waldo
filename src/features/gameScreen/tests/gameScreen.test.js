import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import { toMatchDiffSnapshot } from 'snapshot-diff';
import GameScreen from "../gameScreen";

expect.extend({ toMatchDiffSnapshot });

test("GameScreen is rendered correctly", () => {
  const { container } = render(<GameScreen />);

  expect(container).toMatchSnapshot();
});

jest.mock("../../onBoardScreen/character", () => ({ characterName }) => {
  return <p>{characterName}</p>;
});

test("clicking the circle with 3 written in it, shows a tooltip of characters", async() => {
  const user = userEvent.setup();
  const {asFragment} = render(<GameScreen/>);
  const firstRender = asFragment();

  const circle = screen.getByText(/3/i);
  await user.click(circle);

  expect(firstRender).toMatchDiffSnapshot(asFragment())
});
