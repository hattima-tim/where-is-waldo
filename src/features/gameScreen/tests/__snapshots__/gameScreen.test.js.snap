// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameScreen is rendered correctly 1`] = `
<div>
  <div>
    <header
      class="flex h-16 items-center justify-around bg-[#0e0c31]"
    >
      <h1
        class="text-2xl font-bold text-white"
      >
        Where's 
        <span
          class="text-[#ff0000]"
        >
          Waldo
        </span>
      </h1>
      <div
        class="text-2xl font-bold text-white"
        role="timer"
      >
        00.00.00
      </div>
      <div>
        <div
          class="flex h-12 w-12 items-center justify-center rounded-full bg-[#8d0c0c] text-2xl text-white"
        >
          3
        </div>
      </div>
    </header>
    <img
      alt="universe-113"
      src="https://res.cloudinary.com/du3oueesv/image/upload/v1672759136/Where%27s%20Waldo/universe-113.0d4fe7c1_pfzqhw.jpg"
    />
  </div>
</div>
`;

exports[`clicking the circle with 3 written in it, shows a tooltip of characters 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -23,10 +23,23 @@
          <div
            class=\\"flex h-12 w-12 items-center justify-center rounded-full bg-[#8d0c0c] text-2xl text-white\\"
          >
            3
          </div>
+         <div
+           class=\\"absolute right-40 z-10 w-72 rounded-md bg-[#17134d] p-4 text-white transition-all\\"
+         >
+           <p>
+             Cacodemon
+           </p>
+           <p>
+             Bill Cipher
+           </p>
+           <p>
+             Courage
+           </p>
+         </div>
        </div>
      </header>
      <img
        alt=\\"universe-113\\"
        src=\\"https://res.cloudinary.com/du3oueesv/image/upload/v1672759136/Where%27s%20Waldo/universe-113.0d4fe7c1_pfzqhw.jpg\\""
`;
